packages/entando-apimanager/modules/api/apiManager.js

```diff
@@ -133,11 +133,37 @@ const getRequestParams = (request) => {
   return requestParams;
 };
 
+const isAbsoluteUrl = (url) => {
+  try {
+    new URL(url); // eslint-disable-line no-new
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
 const getCompleteRequestUrl = (request, params = {}) => {
+  // note on new URL():
+  // if url (first parameter) is an absolute path, base domain is not used
+  // if base domain is provided, it has to be absolute (even when url is absolute!)
+  // e.g.,
+  //  new URL('http://www.entando.com') -> { href: 'http://www.entando.com/', ... }
+  //  new URL('/digital-exchange', 'http://www.entando.com') -> { href: 'http://www.entando.com/digital-exchange', ... }
+  //  new URL('/digital-exchange') -> error [ERR_INVALID_URL]: Invalid URL: /digital-exchange
+  //  new URL('http://www.entando.com', 'not-url') -> error [ERR_INVALID_URL]: Invalid URL: not-url
+  //  new URL('http://www.entando.com', '/digital-exchange') -> error [ERR_INVALID_URL]: Invalid URL: /digital-exchange
+
   const storedDomain = getDomain(store.getState());
```

Comment:
can we please provide a name that makes clear that domain value is coming from ENV variable?